### PR TITLE
Improve initialization

### PR DESCRIPTION
**Pull request by hormati**

### CODE REVIEW

File: src/app/main/main.component.ts [hunk 1]
```diff
@@ -166,6 +163,18 @@ export class MainComponent implements OnInit, OnDestroy {
...
+    this.project = this._route.snapshot.paramMap.get("project");
+    this.dataset = this._route.snapshot.paramMap.get("dataset");
+    this.table = this._route.snapshot.paramMap.get("table");
+    this.jobID = this._route.snapshot.paramMap.get("job");
+    this.location = this._route.snapshot.paramMap.get("location");
+    // To set the dropdown to "Auto Select", we need empty string for location.
+    if (!this.location) {
+      this.location = '';
+    }
+  }
```
Comment (donmccurdy) [inline src/app/main/main.component.ts]: While it's verbose, I think I do prefer having these properties all set in ngOnInit.

Comment (hormati) [reply src/app/main/main.component.ts]: done.

File: src/app/main/main.component.ts [hunk 2]
```diff
@@ -210,18 +215,24 @@ export class MainComponent implements OnInit, OnDestroy {
         });
 
       if (this._hasJobParams() && this._jobParamsValid()) {
+        this.dataFormGroup.patchValue({
+          sql: '/* Loading sql query from job... */',
+          projectID: this.project,
+          location: this.location
+        });
         this.dataService.getQueryFromJob(this.jobID, this.location, this.project).then((queryText) => {
           this.queryTextFromJob = queryText.sql;
```
Comment (donmccurdy) [inline src/app/main/main.component.ts]: It doesn't look like `this.queryTextFromJob` is being used anywhere, can we just keep it a local variable?

File: src/app/main/main.component.ts [hunk 3]
```diff
@@ -210,18 +215,24 @@ export class MainComponent implements OnInit, OnDestroy {
...
           this.dataFormGroup.patchValue({
             sql: this.queryTextFromJob,
-            projectID: this.project
+            projectID: this.project,
+            location: this.location
```
Comment (donmccurdy) [inline src/app/main/main.component.ts]: I don't think you have to specify projectID and location again here — just:

```js
this.dataFormGroup.patchValue({sql: queryText.sql});
```

File: src/app/main/main.component.ts [hunk 4]
```diff
@@ -210,18 +215,24 @@ export class MainComponent implements OnInit, OnDestroy {
...
           });
         });
       } else if (this._hasTableParams() && this._tableParamsValid()) {
         this.dataFormGroup.patchValue({
           sql: `SELECT * FROM \`${this.project}.${this.dataset}.${this.table}\`;`,
           projectID: this.project,
-          location: this.location
+          location: ''
```
Comment (donmccurdy) [inline src/app/main/main.component.ts]: Do we need to clear `location`, or could we just not patch it here?

File: src/app/main/main.component.ts [hunk 5]
```diff
@@ -132,11 +132,16 @@ export class MainComponent implements OnInit, OnDestroy {
...
+    // Read parameters from URL
     this.project = this._route.snapshot.paramMap.get("project");
     this.dataset = this._route.snapshot.paramMap.get("dataset");
     this.table = this._route.snapshot.paramMap.get("table");
     this.jobID = this._route.snapshot.paramMap.get("job");
     this.location = this._route.snapshot.paramMap.get("location");
+    // To set the dropdown to "Auto Select", we need empty string for location.
+    if (!this.location) {
+      this.location = '';
+    }
```
Comment (donmccurdy) [inline src/app/main/main.component.ts]: For brevity, how about:

```
this.location = this._route.snapshot.paramMap.get('location') || '';  // Empty string for 'Auto Select'.
```

Comment (hormati) [reply src/app/main/main.component.ts]: done

File: src/app/main/main.component.ts (hunk 4 above)
Comment (hormati) [reply src/app/main/main.component.ts]: Not patching it here is better.

File: src/app/main/main.component.ts (hunk 3 above)
Comment (hormati) [reply src/app/main/main.component.ts]: Done.

File: src/app/main/main.component.ts (hunk 2 above)
Comment (hormati) [reply src/app/main/main.component.ts]: Yup. good catch.

**Pull request merged**